File: fortran03/ch01/test_table_search.f90
program test_table_search

    use table_search
    use test

    implicit none

    print *, "Testing..."

    print *, ""
    print *, "table_search_search()"
    call test_table_search_search()

contains

    subroutine test_table_search_search()
        character(len=6), dimension(5) :: table

        table = ["best  ", "fest  ", "guest ", "test  ", "wriest"]

        call test_assert_equal( table_search_search(table,  "chest" ), 5, "test 01" )
        call test_assert_equal( table_search_search(table,  "best"  ), 0, "test 02" )
        call test_assert_equal( table_search_search(table,  "test"  ), 3, "test 03" )
        call test_assert_equal( table_search_search(table,  "wriest"), 4, "test 04" )
        call test_assert_equal( table_search_search(table,  "wries" ), 5, "test 05" )
    end subroutine test_table_search_search

end program test_table_search

File: c99/ch01/test_table_search.c
#include <stdio.h>
#include <string.h>

#include "table_search.h"
#include "test.h"

void test_table_search_search(void);

int main(void)
{
    printf("Testing...\n\n");

    printf("table_search_search()\n");
    test_table_search_search();

    return 0;
}

void test_table_search_search(void)
{
    #define COUNT 5
    const char* table[COUNT] = { "best", "fest", "guest", "test", "wriest" };

    test_require_equal_size_t(
        table_search_search(table, COUNT, "chest"), COUNT, "test 01"
    );

    test_require_equal_size_t(
        table_search_search(table, COUNT, "best"), 0, "test 02"
    );

    test_require_equal_size_t(
        table_search_search(table, COUNT, "test"), 3, "test 03"
    );

    test_require_equal_size_t(
        table_search_search(table, COUNT, "wriest"), 4, "test 04"
    );

    test_require_equal_size_t(
        table_search_search(table, COUNT, "wries"), COUNT, "test 05"
    );
}
